### PR TITLE
Update `broadcast_expand`, `sign_expand`, `fold_global`, and `mitigate_with_zne`

### PR DESCRIPTION
**Pull request by eddddddy**
**Context:**
Update the batch transforms to the new transform API.

**Description of the Change:**
Update `broadcast_expand`, `sign_expand`, `fold_global`, and `mitigate_with_zne`.

### CODE REVIEW

File: pennylane/transforms/mitigate.py [hunk 1]
```diff
@@ -520,6 +517,11 @@ def circuit(w1, w2):
     ]
 
     tapes = [tape_ for tapes_ in tapes for tape_ in tapes_]  # flattens nested list
+
+    # if folding was a batch transform, ignore the processing function
+    if isinstance(tapes[0], tuple) and isinstance(tapes[0][0], list) and callable(tapes[0][1]):
```
Comment (rmoyard) [inline pennylane/transforms/mitigate.py:522]: Can you explain a bit more what is happening here?

**Review by rmoyard — COMMENTED** [review]
It looks good! Just one question about folding in the transform ZNE

File: pennylane/transforms/mitigate.py (hunk 1 above)
Comment (eddddddy) [reply pennylane/transforms/mitigate.py:522]: In this function, `folding` is expected to be a callable that takes in a tape and returns another tape. It is also possible to pass in a batch transform, where only the first tape in the list of tapes is considered and the rest is ignored. This logic was previously handled by lines 511-514 (now deleted).

This change maintains the same behaviour as before, while also allowing for other batch transforms to be passed in. (Previously only `fold_global` works correctly.)

Comment (rmoyard) [reply pennylane/transforms/mitigate.py:522]: Thanks !

**Review by rmoyard — APPROVED** [review]

**Pull request merged**